files/default/neutron-ha-tool.py

```diff
@@ -598,15 +598,14 @@ def migrate_l3_routers_from_agent(qclient, agent, targets, agent_picker,
                                   router_filter, noop, wait_for_router,
                                   delete_namespace):
     LOG.info("Querying agent_id=%s for routers to migrate away", agent['id'])
-    router_id_list = list_routers_on_l3_agent(qclient, agent['id'])
-    router_id_list = router_filter.filter_routers(router_id_list)
```

Comment:
very nice catch. I won't change the router filter to deal with routers instead of router ids since it writes the ids to a file. Instead I'll use some list comprehensions to get the same behavior.